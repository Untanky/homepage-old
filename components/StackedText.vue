<template lang="pug">
  .stacked-text.flex.flex-col
    div(v-for="formattedText, index in formattedTexts" :key="index" :class="formattedText.classes") {{ formattedText.text }}
</template>

<script lang="ts">
import Vue, { PropType } from 'vue';
import FormattedText from '../src/FormattedText';

export default Vue.extend({
  name: 'StackedText',
  props: {
    formattedTexts: {
      type: Array as PropType<Array<FormattedText>>,
      required: true,
    },
  },
});
</script>
