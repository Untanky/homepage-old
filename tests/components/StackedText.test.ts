/* eslint-disable no-undef */
// global it, expect
import { mount } from '@vue/test-utils';
import StackedText from '../../components/StackedText.vue';

it('should run', () => {
  const props = {
    data: {
      formattedTexts: [
        {
          text: 'Hello World',
          classes: ['font-bold', 'text-xl'],
        },
        {
          text: 'Goodbye World',
          classes: ['italic', 'text-xl'],
        },
      ],
      classes: ['justify-between'],
    },
  };

  const comp = mount(StackedText, { propsData: props });
  expect(comp.element).toMatchSnapshot();
});
