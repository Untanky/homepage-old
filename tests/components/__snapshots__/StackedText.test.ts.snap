// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should run 1`] = `
<div
  class="stacked-text flex row"
>
  <div
    class="bold text-xl"
  >
    formattedText.text
  </div>
  <div
    class="italic text-xl"
  >
    formattedText.text
  </div>
</div>
`;
